c++/src/kj/compat/tls.c++

```diff
@@ -80,6 +90,33 @@ void throwOpensslError() {
   kj::throwFatalException(getOpensslError());
 }
 
+#ifdef _WIN32
+// Adds windows CA store certificates to OpenSSL's ca store
+// Taken from https://stackoverflow.com/questions/9507184/can-openssl-on-windows-use-the-system-certificate-store
+void updateOpenSSLCAStoreWithWindowsCertificates(SSL_CTX* ctx) {
+  X509_STORE* store = SSL_CTX_get_cert_store(ctx);
+  if (store == nullptr) {
+    throwOpensslError();
+  }
+  HCERTSTORE hStore = CertOpenSystemStoreW(NULL, L"ROOT");
+  KJ_ASSERT(hStore != nullptr, "Failed to open system certificate store");
+  
+  PCCERT_CONTEXT pContext = nullptr;
+  while ((pContext = CertEnumCertificatesInStore(hStore, pContext))) {
+    X509* x509 = d2i_X509(nullptr, (const unsigned char**)&pContext->pbCertEncoded, pContext->cbCertEncoded);
+    if (x509) {
+      if (!X509_STORE_add_cert(store, x509)) {
+        throwOpensslError();
```

Comment:
Looks like if this throws we'll leak some memory as various cleanup below will be skipped. Maybe use KJ_DEFER() to register all the cleanup steps so they happen either way?

Reply:
Done